test/SmartCDNNodeParityTest.php

```diff
@@ -0,0 +1,231 @@
+      'input' => $this->input,
+      'expire_at_ms' => $this->expireAt,
+      'url_params' => [
+        'width' => null,
+        'height' => 200
+      ]
+    ];
+
+    $url = $this->transloadit->signedSmartCDNUrl(
+      $params['workspace'],
+      $params['template'],
+      $params['input'],
+      $params['url_params'],
+      ['expireAtMs' => $params['expire_at_ms']]
+    );
+    $nodeUrl = $this->runNodeScript($params);
+
+    $this->assertEquals($nodeUrl, $url, 'Null values in parameters should be handled the same as node');
+  }
+
+  public function testEmptyStringValuesInParameters(): void {
+    $params = [
+      'workspace' => $this->workspace,
+      'template' => $this->template,
+      'input' => $this->input,
+      'expire_at_ms' => $this->expireAt,
+      'url_params' => [
+        'width' => '',
+        'height' => 200
+      ]
+    ];
+
+    $url = $this->transloadit->signedSmartCDNUrl(
+      $params['workspace'],
+      $params['template'],
+      $params['input'],
+      $params['url_params'],
+      ['expireAtMs' => $params['expire_at_ms']]
+    );
+    $nodeUrl = $this->runNodeScript($params);
+
+    $this->assertEquals($nodeUrl, $url, 'Empty string values in parameters should be handled the same as node');
+  }
+
+  public function testArrayValuesInParameters(): void {
+    $params = [
+      'workspace' => $this->workspace,
+      'template' => $this->template,
+      'input' => $this->input,
+      'expire_at_ms' => $this->expireAt,
+      'url_params' => [
+        'tags' => ['landscape', 'amsterdam', null, ''],
+        'height' => 200
+      ]
+    ];
+
+    $url = $this->transloadit->signedSmartCDNUrl(
+      $params['workspace'],
+      $params['template'],
+      $params['input'],
+      $params['url_params'],
+      ['expireAtMs' => $params['expire_at_ms']]
+    );
+    $nodeUrl = $this->runNodeScript($params);
+
+    $this->assertEquals($nodeUrl, $url, 'Array values in parameters should be handled the same as node');
+  }
+
+  public function testSpecialCharactersInPaths(): void {
+    $params = [
+      'workspace' => 'my workspace',
+      'template' => 'template/with/slashes',
+      'input' => 'input with spaces.jpg',
+      'expire_at_ms' => $this->expireAt
+    ];
+
+    $url = $this->transloadit->signedSmartCDNUrl(
+      $params['workspace'],
+      $params['template'],
+      $params['input'],
+      [],
+      ['expireAtMs' => $params['expire_at_ms']]
+    );
+    $nodeUrl = $this->runNodeScript($params);
+
+    $this->assertEquals($nodeUrl, $url, 'URLs with spaces and special characters should match node implementation');
+  }
+
+  public function testEmptyInputField(): void {
+    $params = [
+      'workspace' => 'workspace',
+      'template' => 'template',
+      'input' => 'minimal.jpg',
+      'expire_at_ms' => $this->expireAt
+    ];
+
+    $url = $this->transloadit->signedSmartCDNUrl(
+      $params['workspace'],
+      $params['template'],
+      $params['input'],
```

Comment:
The test name suggests that this logic covers situations where `input` might be empty, but as far as I can tell `$params['input']` is always non-empty.

Reply:
Empty string input is now allowed